casepro/msgs/models.py

```diff
@@ -266,13 +266,16 @@ def as_json(self, full=True):
         if full:
             if not self.parent:
                 parent_json = None
+                result['labels'] = [l.as_json() for l in self.labels.all()]
             else:
                 parent_json = self.parent.id
 
+                labels = FAQ.objects.prefetch_related('labels').get(pk=parent_json).labels.all()
```

Comment:
This doesn't help because you're still separately fetching parent labels for each FAQ. The idea is to prefetch somewhere like `FAQ.get_all()` so that only one db hit is used to fetch parent labels for all the FAQs being fetched.

Best way to test stuff like this is fetch and serialize a bunch of FAQs together and use `with self.assertNumQueries(N):` to check that you're not making db queries for each FAQ being returned.